server/src/Auth.js

```diff
@@ -0,0 +1,84 @@
+import crypto from 'crypto';
+
+class Auth {
+    checkUsername(req, res, AuthModel) {
+        const username = req.body.username.toLowerCase();
+        console.log(`Checking username ${username}`);
+
+        AuthModel.find({ username: username}, (err, arr) => {
+            if (arr.length > 0) {
+                console.log(`${username} exists.`);
+                res.json({exists: true});
+            } else {
+                console.log(`${username} not found.`);
+                res.json({exists: false});
+            }
+        });
+    }
+
+    checkLogin(req, res, AuthModel) {
+        const login = {
+            username: req.body.username.toLowerCase(),
+            password: req.body.password,
+        };
+
+        console.log('Checking Login');
+
+        AuthModel.find({ username: login.username}, (err, arr) => {
+            if (err) throw err;
+            this._verifyLogin(arr, login, res);
+        });
+    }
+
+    register(req, res, AuthModel, UserModel) {
+        //const collection = db.collection(this.collectionStr);
+        const login = {
+            username: req.body.username.toLowerCase(),
+            password: req.body.password,
```

Comment:
Do you want to add any password requirements?  I think a minimum number of characters would be a good idea at least.